src/rules/no-if.ts

```diff
@@ -1,36 +1,54 @@
-import { getDocsUrl, isTestCase } from './util';
+import { TestCaseName, createRule, getNodeName, isTestCase } from './tsUtils';
+import {
+  AST_NODE_TYPES,
+  TSESTree,
+} from '@typescript-eslint/experimental-utils';
 
-const isTestArrowFunction = node =>
+const testCaseNames = new Set<string | null>([
```

Comment:
null? seems wrong

Reply:
Yeah that's just b/c of how the typings work in relation to `#has`.

Basically, it's defined like this:

```
interface Set<T> {
  has(item: T): boolean;
}
```

That means if `T` is just `string`, you can't do `has` checks against things like `null`, which is a possible return of `getNodeName` :)

Originally I had `getNodeName(/* ... */) || ''`, but since that's a runtime solution, it's expected to be tested for by branch coverage :joy: